src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateMethodSymbol.cs

```diff
@@ -97,12 +98,18 @@ protected void InitializeParameters(ImmutableArray<ParameterSymbol> parameters)
                 diagnostics.Add(ErrorCode.ERR_BadVisDelegateReturn, delegateType.Locations[0], delegateType, invoke.ReturnType);
             }
 
-            foreach (var parameter in invoke.Parameters)
+            for (int i = 0; i < invoke.Parameters.Length; i++)
             {
-                if (!parameter.TypeWithAnnotations.IsAtLeastAsVisibleAs(delegateType, ref useSiteDiagnostics))
+                var parameterSymbol = invoke.Parameters[i];
+                if (!parameterSymbol.TypeWithAnnotations.IsAtLeastAsVisibleAs(delegateType, ref useSiteDiagnostics))
                 {
                     // Inconsistent accessibility: parameter type '{1}' is less accessible than delegate '{0}'
-                    diagnostics.Add(ErrorCode.ERR_BadVisDelegateParam, delegateType.Locations[0], delegateType, parameter.Type);
+                    diagnostics.Add(ErrorCode.ERR_BadVisDelegateParam, delegateType.Locations[0], delegateType, parameterSymbol.Type);
+                }
+                var parameterSyntax = syntax.ParameterList.Parameters[i];
+                if (parameterSyntax.ExclamationToken.Kind() == SyntaxKind.ExclamationToken)
```

Comment:
Down the line you may want a property `bool SourceParameterSymbol.IsNullChecked { get; }`

Reply:
Asked Jared if Symbol changes should be in this PR when I'd started and he said it should be the next one.